kotlinlib/src/mill/kotlinlib/KotlinModule.scala

```diff
@@ -102,19 +102,48 @@ trait KotlinModule extends JavaModule { outer =>
     )().toSeq ++ kotlinWorkerClasspath()
   }
 
+  /** Flag to use the embeddable kotlin compiler */
+  def kotlinCompilerEmbeddable: Boolean = false
+
+  /**
+   * If ksp plugins are used, switch to embeddable to avoid
+   * any classpath conflicts.
+   *
+   * @return kotlin-compiler or kotlin-compiler-embeddable dependency
+   */
```

Comment:
The mention of `ksp` is misleading here.

```suggestion
  /**
   * The kotlin-compiler dependency. It uses the embeddable version, if [[kotlinCompilerEmbeddable]] is `true`.
   */
```